### PR TITLE
SCHEMA: Create context schema in meta/

### PR DESCRIPTION
**Pull request by effigies**
Builds on @rwblair's work in #1059.

### CODE REVIEW

File: src/schema/meta/context.yaml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+# The subject namespace is constructed once per subject, and is available when
+# visiting all files within that subject.
+#
+# All other (current) namespaces are defined on individual files.
+# Sidecar metadata and file associations are built according to the inheritance
+# principle.
+#
+---
+context:
+    dataset:
```
Comment (tsalo) [inline src/schema/meta/context.yaml:21]: I believe we also use `dataset.modalities`, but I guess we might want to access that through `subject` or `session`...

File: src/schema/meta/context.yaml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+            type: str
+        modality:
+            description: "Modality of current file, e.g., MRI"
+            type: str
+        sidecar:
+            description: "Sidecar metadata constructed via the inheritance principle"
+            type: map
+        associations:
+            description: |
+                Associated files, indexed by suffix, selected according to the inheritance principle
```
Comment (tsalo) [inline src/schema/meta/context.yaml:103]: Should this include other files in a file collection, such as the other `echo-*` files for a ME-EPI file?

Comment (tsalo) [inline src/schema/meta/context.yaml:103]: Should this include other files in a file collection, such as the other `echo-*` files for a ME-EPI file?

Comment (tsalo) [inline src/schema/meta/context.yaml:103]: Should this include other files in a file collection, such as the other `echo-*` files for a ME-EPI file?

File: src/schema/meta/context.yaml [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+            type: map
+            properties:
+                events:
+                    description: "Path to associated events file"
+                    type: str
+                aslcontext:
+                    description: "Path to associated aslcontext file"
+                    type: str
```
Comment (tsalo) [inline src/schema/meta/context.yaml:111]: Do we want to hardcode associations in this file? There are a lot of them, and they all depend on the original file's suffix and/or entities.

Comment (tsalo) [inline src/schema/meta/context.yaml:111]: Do we want to hardcode associations in this file? There are a lot of them, and they all depend on the original file's suffix and/or entities.

Comment (tsalo) [inline src/schema/meta/context.yaml:111]: Do we want to hardcode associations in this file? There are a lot of them, and they all depend on the original file's suffix and/or entities.

Comment (effigies) [reply src/schema/meta/context.yaml]: @tsalo See latest file. I try to make clear that these are only ones that are used in rules.

File: src/schema/meta/context.yaml (hunk 2 above)
Comment (effigies) [reply src/schema/meta/context.yaml:113]: Are there rules that this would help us enforce?

**Pull request merged**